narwhals/_expression_parsing.py

```diff
@@ -223,45 +234,57 @@ def expansion_kind(self) -> ExpansionKind:
     def with_kind(self, kind: ExprKind, /) -> ExprMetadata:
         """Change metadata kind, leaving all other attributes the same."""
         return ExprMetadata(
-            kind, n_open_windows=self._n_open_windows, expansion_kind=self._expansion_kind
+            kind,
+            n_opened_windows=self._n_opened_windows,
+            n_closed_windows=self._n_closed_windows,
+            expansion_kind=self._expansion_kind,
         )
 
     def with_extra_open_window(self) -> ExprMetadata:
-        """Increment `n_open_windows` leaving other attributes the same."""
+        """Increment `n_opened_windows` leaving other attributes the same."""
         return ExprMetadata(
             self.kind,
-            n_open_windows=self._n_open_windows + 1,
+            n_opened_windows=self._n_opened_windows + 1,
             expansion_kind=self._expansion_kind,
+            n_closed_windows=self._n_closed_windows,
         )
 
     def with_kind_and_extra_open_window(self, kind: ExprKind, /) -> ExprMetadata:
-        """Change metadata kind and increment `n_open_windows`."""
+        """Change metadata kind and increment `n_opened_windows`."""
         return ExprMetadata(
             kind,
-            n_open_windows=self._n_open_windows + 1,
+            n_opened_windows=self._n_opened_windows + 1,
+            n_closed_windows=self._n_closed_windows,
             expansion_kind=self._expansion_kind,
         )
 
     @staticmethod
-    def simple_selector() -> ExprMetadata:
+    def selector_simple() -> ExprMetadata:
```

Comment:
https://github.com/narwhals-dev/narwhals/pull/2351#discussion_r2029968131

Apologies if you noticed this but preferred **simple**

> | Current                         | Proposed                 |
> | ------------------------------- | ------------------------ |
> | `simple_selector`               | `selector_single`        |
> 
> Also the **suffix matches** the `ExpansionKind` name

```suggestion
    def selector_single() -> ExprMetadata:
```

```py
ExpansionKind.SINGLE
```

Reply:
ah yup, thanks!